### PR TITLE
Parity Blockchain/Wallet clients/services implementation

### PR DESCRIPTION
**Pull request by ybondur**

### CODE REVIEW

**Review by raidray — APPROVED** [review]

**Comment by raidray** [comment]
@mod when this files will merge with master ?

**Comment by mnaichuk** [comment]
Same for 1-9, please

File: app/models/wallet.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@
 # frozen_string_literal: true
 
 
+
```
Comment (ysv) [inline app/models/wallet.rb]: Leave sing blank line here

File: app/services/blockchain_service/geth.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+
+module BlockchainService
```
Comment (ysv) [inline app/services/blockchain_service/geth.rb:4]: We need migration where we update all blockchains with ethereum client to geth

File: config/templates/seed/blockchains.yml.erb [hunk 3]
```diff
@@ -1,9 +1,20 @@
...
+  min_confirmations: 6                      # Minimal confirmations needed for withdraw and deposit confirmation.
+  explorer:
+    address:         https://kovan.etherscan.io/address/#{address}
+    transaction:     https://kovan.etherscan.io/tx/#{txid}
+  status:            disabled
+
 - key:               eth-rinkeby
   name:              Ethereum Rinkeby
-  client:            ethereum               # API client name.
+  client:            geth               # API client name.
```
Comment (ysv) [inline config/templates/seed/blockchains.yml.erb]: align # API client name.

File: lib/wallet_client/parity.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+
+module WalletClient
+  class Parity < Ethereum
+    def create_eth_withdrawal!(issuer, recipient, amount, options = {})
```
Comment (ysv) [inline lib/wallet_client/parity.rb]: This code is outdated
In a new version we require gas_limit and gas_price

I'm not even sure you need to override this methods

File: spec/services/blockchain_service/geth_spec.rb [hunk 5]
```diff
@@ -323,6 +323,7 @@ def request_body(block_number, index)
 
       it 'changes withdraw confirmations amount' do
         subject.each do |withdrawal|
+          # binding.pry
```
Comment (ysv) [inline spec/services/blockchain_service/geth_spec.rb]: Remove this comment

File: app/services/wallet_service/ethereum.rb [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    end
+
+    def deposit_collection_fees(deposit, options = {})
+      currency_options = deposit.currency.options.symbolize_keys
+
+      # Calculate fees paid in gas for deposit collection of ERC20.
+      # Use currency gas_limit and gas_price options if they are present.
+      # Otherwise use DEFAULT_ERC20_FEE.
+      eth_fees_value =
+        if currency_options.key?(:gas_limit) && currency_options.key?(:gas_price)
```
Comment (ysv) [inline app/services/wallet_service/ethereum.rb]: @mnaichuk we need to fix issue you have faced here

**Review by ysv — CHANGES_REQUESTED** [review]
Also add comment to Ethereum services and clients that they are abstract

**Pull request merged**